README.md

```diff
@@ -130,6 +130,54 @@ data List = (List.cons head tail) | (List.nil)
 ListEx2 = (List.cons 1 (List.cons 2 (List.cons 3 List.nil)))
 ```
 
+It's possible to match different kinds of terms. These three forms are equivalent:
+```rs
+match list {
+  (List.cons hd tl):  (Some hd)
+  List.nil:  None
+}
+
+match list {
```

Comment:
can you add a comment here explaining that if we don't provide binds to the fields, then it uses implicit binding using `(Ctr arg.field1 arg.field2)`?